var/spack/repos/builtin/packages/miopen-hip/package.py

```diff
@@ -27,20 +29,36 @@ class MiopenHip(CMakePackage):
     depends_on('bzip2', type='link')
     depends_on('sqlite', type='link')
     depends_on('half', type='build')
-    for ver in ['3.5.0', '3.7.0', '3.8.0']:
+
+    patch('0001-Add-rocm-path-and-rocm-device-lib-path-flags.patch', when='@3.9.0:')
+
+    for ver in ['3.5.0', '3.7.0', '3.8.0', '3.9.0', '3.10.0']:
         depends_on('hip@' + ver, type='build', when='@' + ver)
         depends_on('rocm-cmake@' + ver, type='build', when='@' + ver)
         depends_on('comgr@' + ver, type='link', when='@' + ver)
         depends_on('llvm-amdgpu@' + ver, type='build', when='@' + ver)
         depends_on('rocm-clang-ocl@' + ver, type='build', when='@' + ver)
         depends_on('rocblas@' + ver, type='link', when='@' + ver)
         depends_on('rocm-device-libs@' + ver, type='link', when='@' + ver)
+        if ver in ['3.9.0:']:
+            depends_on('zlib', type=('link'))
+
+    def setup_build_environment(self, build_env):
+        if '@3.9.0:' in self.spec:
+            lib_dir = self.spec['zlib'].prefix.lib
+            build_env.prepend_path('LIBRARY_PATH', lib_dir)
```

Comment:
Spack automatically adds link deps to the compiler wrapper, is this needed?

Reply:
I Get the below error if i don't add this . its not able to find the library. So, I am guessing this may be required.

` /usr/lib64/gcc/x86_64-suse-linux/7/../../../../x86_64-suse-linux/bin/ld: cannot find -lz
  >> 377    clang-12: error: linker command failed with exit code 1 (use -v to see invocation)
  >> 378    make[2]: *** [src/CMakeFiles/MIOpen.dir/build.make:2499: lib/libMIOpen.so.1.0] Error 1
     379    make[2]: Leaving directory '/tmp/root/spack-stage/spack-stage-miopen-hip-3.9.0-77jpdcmrxfrivgqdrdy6yuclohxhozrp/spack-build-77jpdcm'
  >> 380    make[1]: *** [CMakeFiles/Makefile2:1980: src/CMakeFiles/MIOpen.dir/all] Error 2
     381    make[1]: Leaving directory '/tmp/root/spack-stage/spack-stage-miopen-hip-3.9.0-77jpdcmrxfrivgqdrdy6yuclohxhozrp/spack-build-77jpdcm'
     382    make: *** [Makefile:185: all] Error 2
`

Reply:
Hmm, must not be using the compiler wrappers then. Can you rename `build_env` to `env` for consistency?

Reply:
Ok. Changing that.